server/src/test/java/se/fortnox/reactivewizard/server/AbortedRequestHandlingTest.java

```diff
@@ -0,0 +1,91 @@
+package se.fortnox.reactivewizard.server;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.log4j.Appender;
+import org.apache.log4j.Level;
+import org.apache.log4j.LogManager;
+import org.junit.Test;
+import reactor.core.publisher.Flux;
+import reactor.netty.http.client.HttpClient;
+import rx.Observable;
+import se.fortnox.reactivewizard.ExceptionHandler;
+import se.fortnox.reactivewizard.RequestHandler;
+import se.fortnox.reactivewizard.jaxrs.JaxRsRequestHandler;
+import se.fortnox.reactivewizard.jaxrs.JaxRsResourceFactory;
+import se.fortnox.reactivewizard.test.LoggingMockUtil;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static rx.Observable.just;
+import static se.fortnox.reactivewizard.test.TestUtil.matches;
+
+/**
+ * When client aborts the connection the server should still finish gracefully with only a debug-log as the result
+ */
+public class AbortedRequestHandlingTest {
+
+    @Test
+    public void shouldHandleAbortedRequestGracefully() throws NoSuchFieldException, IllegalAccessException {
+
+        final Appender mockedLogAppender = LoggingMockUtil.createMockedLogAppender(ExceptionHandler.class);
+        LogManager.getLogger(ExceptionHandler.class).setLevel(Level.DEBUG);
+
+        RwServer      rwServer                   = null;
+        AtomicBoolean serverFinishedWithoutError = new AtomicBoolean(false);
+
+        try {
+            JaxRsRequestHandler jaxRsRequestHandler = new JaxRsRequestHandler(new Object[]{new TestResourceImpl()}, new JaxRsResourceFactory(), new ExceptionHandler(), false);
+
+            rwServer = server((httpServerRequest, httpServerResponse) -> {
+                httpServerRequest.withConnection(connection -> {
+                    connection.channel().close();
```

Comment:
This is closing the connection on the server side, but the problem that we see is that the client closes the connection. Perhaps it results in the same error, but I think it is a different scenario. Maybe the client closing is harder to simulate due to http keepalive and stuff.

Have you tried just createing a server with a delay in the response, and have a client with a really short timeout set? That might result in the client closing the connection before the server writes the response.

Reply:
Yes, I tried that at first but didn't succeed. I will give it another try.

Yes, we close the connection at the server side but it will generate the same type of exception that is thrown when the client closes the connection. 
I think who closes the connection is not the important thing but rather that the client connection is closed when trying to send data.